lib/_http_client.js

```diff
@@ -68,6 +68,11 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
+  const _method = options.method;
+  if (_method !== undefined && _method !== null
+    && typeof _method !== 'string') {
+    throw new TypeError('Method must be a string');
+  }
```

Comment:
Well, it's a difference between semver-ness. This change is definitely going to be semver-major as-is, but it could be semver-patch with the check I suggested. It's up to you.